apps/src/lab2/lab2Redux.ts

```diff
@@ -123,6 +126,18 @@ export const setUpWithLevel = createAsyncThunk(
         return;
       }
 
+      // Start mode doesn't use channel ids so we can skip creating
+      // a project manager and just set the level data.
+      const appOptions = getScriptData('appoptions') as PartialAppOptions;
```

Comment:
mind moving the app options related logic into [lab2/utils.ts](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lab2/projects/utils.ts)? We have similar methods there. Then we can have a `isStartMode` method that's a little cleaner.

Reply:
Created a `getAppOptionsEditBlocks` that returns the string if it's available. Also created a `START_SOURCES` constant for use in the components. This seemed to be more useful than just creating `getAppOptionsStartSources` or something, since Lab2 will likely need to work with other modes at some point too.